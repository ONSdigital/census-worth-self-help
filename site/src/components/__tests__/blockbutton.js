import React from "react"
import renderer from "react-test-renderer"
import BlockButton from "../blockbutton"
import { render, fireEvent } from "react-testing-library"

describe("BlockButton", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<BlockButton title="title" subtitle="anything" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("can be clicked", () => {
  	let clickFunction = jest.fn()

  	const { getByTestId } = render(<BlockButton title="title" subtitle="anything" clickFunction={clickFunction}/>);

  	fireEvent.click(getByTestId('block-button'));

    expect(clickFunction).toHaveBeenCalled();
  });
})