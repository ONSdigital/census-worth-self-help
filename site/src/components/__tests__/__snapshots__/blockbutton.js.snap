// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockButton renders correctly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "1vf6ctb",
      "next": undefined,
      "styles": "
        border-radius: 4px;
        box-shadow: 0 2px 4px 0 rgba(212, 212, 212, 0.5);
        border: solid 1px #ececed;
        margin: 0px 10px;
        background-color: rgb(255, 255, 255);
        display: flex;
      ",
    }
  }
  data-testid="block-button"
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "1o8j1iv",
        "next": undefined,
        "styles": "
          
    display: flex;
    justify-content: center;
    flex-direction: column;
  
          margin: 0px 20px;
          color: rgb(0,61,89);
        ",
      }
    }
  >
    <div
      className="Button-heading-Style"
      css={
        Object {
          "map": undefined,
          "name": "km4e7l",
          "next": undefined,
          "styles": "
            text-align: left;
          ",
        }
      }
    >
      title
    </div>
    <div
      className="Card-meta-Style"
      css={
        Object {
          "map": undefined,
          "name": "km4e7l",
          "next": undefined,
          "styles": "
            text-align: left;
          ",
        }
      }
    >
      anything
    </div>
  </div>
</div>
`;
